core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/StreamService.java

```diff
@@ -92,8 +95,28 @@
   }
 
   @PreAuthorize("hasPermission(#stream, 'DELETE')")
-  public void delete(Stream stream) {
-    throw new UnsupportedOperationException();
+  public boolean delete(Stream stream) {
+    StreamKey streamKey = stream.getKey();
+    StreamBinding streamBinding = new StreamBinding(
+            new StreamBindingKey(
+                    streamKey.getDomain(),
+                    streamKey.getName(),
+                    streamKey.getVersion(),
+                    null,
+                    null),
+            null, null);
+    streamBindingService.delete(streamBinding);
```

Comment:
This looks like this deletes the `ConsumerBinding`, `ProducerBinding` and `StreamBinding`. The line below will delete the `Stream`. I _think_ this means we are going to end up with orphaned `Consumer`s and `Producer`s

Reply:
updated to delete consumer and producer along with bindings. assuming 1:1 relationship between producer/producerbinding and consumer/consumerbinding

Reply:
updated to just remove producer/consumer bindings when stream binding is deleted